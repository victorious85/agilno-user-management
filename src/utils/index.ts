export * from './native-alert.ts';
