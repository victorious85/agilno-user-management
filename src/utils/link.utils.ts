import { Linking } from 'react-native';

export const openURL = async (url: string | null) => {
  if (!url) {
    throw new Error(`Wrong url ${url}`);
  }

  try {
    const isUrlOpened = await Linking.openURL(url);
    return isUrlOpened;
  } catch (error) {
    console.error(`Failed to open ${url}`, error);
  }
  return false;
};

export const transformEmail = (email: string) =>
  email ? `mailto:${email}` : null;

export const handleEmailPress = (email: string) =>
  openURL(transformEmail(email));
