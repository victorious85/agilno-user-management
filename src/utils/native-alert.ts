import { Alert, AlertOptions } from 'react-native';

/**
 * Error Alert
 */

interface AppAlert {
  title: string;
  message?: string;
  onPress?: () => void;
  options?: AlertOptions;
}

export const showErrorAlert = (params: AppAlert) =>
  Alert.alert(
    params.title,
    params.message,
    [{ onPress: params?.onPress }],
    params.options,
  );

/**
 * Prompt Alert
 * Allow only one on screen
 */

interface PromptAlert {
  title: string;
  message?: string;
  okButtonTitle?: string;
  cancelButtonTitle?: string;
  onPress?: () => void;
  onPressCancel?: () => void;
  options?: AlertOptions;
  buttonStyle?: 'default' | 'cancel' | 'destructive';
  cancelButtonStyle?: 'default' | 'cancel' | 'destructive';
}

export const showConfirmAlert = (params: PromptAlert) => {
  const {
    title,
    message,
    cancelButtonTitle,
    okButtonTitle,
    onPressCancel = () => {},
    buttonStyle,
    cancelButtonStyle,
  } = params;

  const onPress = () => {
    params?.onPress?.();
  };

  Alert.alert(
    title,
    message,
    [
      {
        text: cancelButtonTitle || 'Cancel',
        onPress: onPressCancel,
        style: cancelButtonStyle || 'default',
      },
      {
        text: okButtonTitle || 'OK',
        onPress: onPress,
        style: buttonStyle || 'destructive',
      },
    ],
    { cancelable: false },
  );
};
