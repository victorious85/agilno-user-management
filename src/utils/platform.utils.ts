import { Platform } from 'react-native';

export const isIos = Platform.OS === 'ios';
export const isAndroid = Platform.OS === 'android';

export const OSVersion = Platform.Version;

export const selectByPlatform = <T, K>(
  iosValue: T,
  androidValue: K,
): T | K | undefined =>
  Platform.select<T | K>({
    ios: iosValue,
    android: androidValue,
  });
