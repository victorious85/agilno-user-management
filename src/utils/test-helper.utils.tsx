import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import { Provider } from 'react-redux';
import store from '../store';

export function findByTestID(testID: string) {
  // @ts-ignore
  return this.find(`[data-testid="${testID}"]`);
}

export function getShallowComponent(Component: any, props: any = {}) {
  const { children = null } = props;

  return ShallowRenderer.createRenderer().render(
    <Component {...props}>{children}</Component>,
  );
}

export function getTreeWithProviders(Component: any, props: any = {}) {
  const { children = null } = props;

  return renderer
    .create(
      <GestureHandlerRootView>
        <SafeAreaProvider>
          <Provider store={store}>
            <Component {...props}>{children}</Component>
          </Provider>
        </SafeAreaProvider>
      </GestureHandlerRootView>,
    )
    .toJSON();
}
