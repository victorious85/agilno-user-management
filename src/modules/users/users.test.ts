// Utils
import { getTreeWithProviders } from '../../utils/test-helper.utils';
// Components
import { Users } from './index';

describe('Users Screen', () => {
  test('renders correctly', () => {
    const component = getTreeWithProviders(Users);
    expect(component).toMatchSnapshot();
  });
});
