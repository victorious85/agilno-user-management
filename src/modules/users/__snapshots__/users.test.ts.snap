// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Screen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
